awx_collection/plugins/module_utils/tower_api.py

```diff
@@ -487,6 +490,9 @@ def delete_if_needed(self, existing_item, on_delete=None):
                 item_name = existing_item['username']
             elif 'identifier' in existing_item:
                 item_name = existing_item['identifier']
+            elif item_type == 'o_auth2_access_token':
```

Comment:
im not sure we'll hit this, testing locally when i output the token object i get:

```
TASK [tower_token : debug] ****************************************************************************************************************************************************************************************************************************************************************************
ok: [testhost] => {
    "msg": {                                                                                                                                                                                                                                                                                          
        "ansible_facts": {                                                                                                                                                                                                                                                                            
            "tower_token": {
                "application": null,                                                                                                                                                                                                                                                                  
                "created": "2020-06-01T18:48:22.790373Z",
                "description": "AWX-Collection-tests-tower_token-description-NtDlNNHIgxVWVYJp",                                                                                                                                                                                                       
                "expires": "3019-10-03T18:48:22.787511Z",
                "id": 461,                                                                                                                                                                                                                                                                            
                "modified": "2020-06-01T18:48:22.799678Z",                                                                                                                                                                                                                                            
                "refresh_token": null,
                "related": {                                                                                                                                                                                                                                                                          
                    "activity_stream": "/api/v2/tokens/461/activity_stream/",
                    "user": "/api/v2/users/1/"
                },
                "scope": "write",
                "summary_fields": {                                                                                                                                                                                                                                                                   
                    "user": {
                        "first_name": "",                                                                                                                                                                                                                                                             
                        "id": 1,
                        "last_name": "",                                                                                                                                                                                                                                                              
                        "username": "admin"
                    }
                },                                                                                                                                                                                                                                                                                    
                "token": "Zvk7PYb0eKusQn6fx0jTgGuvp8Yo9m",                                                                                                                                                                                                                                            
                "type": "o_auth2_access_token",                                                                                                                                                                                                                                                       
                "url": "/api/v2/tokens/461/",
                "user": 1
            }
        },
        "changed": true,
        "failed": false,
        "id": 461,
        "name": "unknown",                                                                                                                                                                                                                                                                            
        "token": "Zvk7PYb0eKusQn6fx0jTgGuvp8Yo9m",
        "warnings": [                                                                                                                                                                                                                                                                                 
            "You are running collection version devel but connecting to tower version 11.2.0"
        ]
    }                                                                                                                                                                                                                                                                                                 
}
```

and if it has a name we use the name, im having trouble locally deleting stuff, im not sure the delete actually works

Reply:
I pushed some updated code for this. Let me know if this resolves the issue.